external-crates/move/crates/move-regex-borrow-graph/src/collections.rs

```diff
@@ -0,0 +1,509 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    bail, ensure, error,
+    references::{Node, Ref},
+    regex::{Extension, Regex},
+    Result,
+};
+use core::fmt;
+use std::collections::{BTreeMap, BTreeSet};
+
+//**************************************************************************************************
+// Definitions
+//**************************************************************************************************
+
+#[derive(Clone, Debug)]
+pub struct Path<Loc, Lbl> {
+    pub loc: Loc,
+    pub(crate) labels: Vec<Lbl>,
```

Comment:
I'm surprised that this doesn't reuse `Regex`

Reply:
I did not want to make it public since it will definitely change

Reply:
This type is only used for borrowed_by and borrows_from